pkg/controller/mysqlcluster/internal/cleaner/pvc.go

```diff
@@ -37,136 +34,129 @@ import (
 		return err
 	}
 
-	recorder.Event(cluster, core.EventTypeNormal, deleteSuccess,
-		fmt.Sprintf(messagePvcDeleted, pvc.Name, cluster.Name))
+	p.recorder.Event(p.cluster, core.EventTypeNormal, reasonPVCCleanupSuccessfull,
+		fmt.Sprintf(messageCleanupSuccessfull, pvc.Name, p.cluster.Name))
 	return nil
 }
 
-func (p *PvcCleaner) getClaims(ctx context.Context, c client.Client) (map[int]*core.PersistentVolumeClaim, error) {
-	meta := p.cluster.ObjectMeta
+func (p *PVCCleaner) getPVCs(ctx context.Context) ([]core.PersistentVolumeClaim, error) {
 	pvcs := &core.PersistentVolumeClaimList{}
 	lo := &client.ListOptions{
-		Namespace:     meta.GetNamespace(),
+		Namespace:     p.cluster.Namespace,
 		LabelSelector: labels.SelectorFromSet(p.cluster.GetLabels()),
 	}
-	err := c.List(ctx, lo, pvcs)
 
-	if err != nil {
+	if err := p.client.List(ctx, lo, pvcs); err != nil {
 		return nil, err
 	}
 
-	return byOrdinal(pvcs), nil
-}
-
-func byOrdinal(allClaims *core.PersistentVolumeClaimList) map[int]*core.PersistentVolumeClaim {
-	claims := map[int]*core.PersistentVolumeClaim{}
-
-	for i := 0; i < len(allClaims.Items); i++ {
-		if allClaims.Items[i].DeletionTimestamp != nil {
-			log.V(2).Info("being deleted, skipping", "pvc", allClaims.Items[i].Name)
-			continue
+	// check just claims with cluster as owner reference
+	claims := []core.PersistentVolumeClaim{}
+	for _, claim := range pvcs.Items {
+		if !isOwnedBy(claim, p.cluster.Unwrap()) {
+			continue // skip it's not owned by this cluster
 		}
 
-		_, ordinal, err := extract(allClaims.Items[i].Name)
-		if err != nil {
-			continue
+		if claim.DeletionTimestamp != nil {
+			continue // is being deleted, skip
 		}
 
-		claims[ordinal] = &allClaims.Items[i]
-
+		claims = append(claims, claim)
 	}
-	return claims
+
+	return claims, nil
 }
 
-func extract(pvcName string) (string, int, error) {
-	idx := strings.LastIndexAny(pvcName, "-")
-	if idx == -1 {
-		return "", 0, fmt.Errorf("PVC does not belong to a StatefulSet")
+func isOwnedBy(pvc core.PersistentVolumeClaim, cluster *api.MysqlCluster) bool {
+	if pvc.Namespace != cluster.Namespace {
+		// check is that cluster is in the same namespace
+		return false
 	}
 
-	name := pvcName[:idx]
-	ordinal, err := strconv.Atoi(pvcName[idx+1:])
-	if err != nil {
-		return "", 0, fmt.Errorf("PVC does not belong to a StatefulSet")
+	for _, ref := range pvc.ObjectMeta.GetOwnerReferences() {
+		if ref.Kind == "MysqlCluster" && ref.Name == cluster.Name {
+			return true
+		}
 	}
-	return name, ordinal, nil
+
+	log.Info("pvc not owner by cluster", "pvc", pvc, "cluster", cluster)
+	return false
 }
 
-func getKeys(m map[int]*core.PersistentVolumeClaim) []int {
-	keys := make([]int, 0, len(m))
-	for k := range m {
-		keys = append(keys, k)
+func getOrdinal(name string) int32 {
+	idx := strings.LastIndexAny(name, "-")
+	if idx == -1 {
+		log.Error(nil, "failed to extract ordinal for pvc", "pvc", name)
+		return -1 // not a good name for PVC
```

Comment:
This might be prone to errors, since you are not explicitly stating the result is an error.
```suggestion
		return -1, errors.New("failed to extract ordinal for pvc") // not a good name for PVC
```